### PR TITLE
[728] apply implement small content improvements

### PR DESCRIPTION
**Pull request by elceebee**
## Context

There was a bug party in October and a few content issues were identified. 

## Changes proposed in this pull request

- Added a few full stops where they were missing 
- Replaced contractions `you're` with `you are` throughout
- Replaced 'you'll' with `you will` throughout 
- Removed ambiguous plurals throughout (eg application(s) is now applications) 
- Replace we'll with we will throughout
- Replace you've with you have throughout
- Replace you'd with you would throughout
- Replace it's with it is throughout
ALSO
- Where tests have been changed, I've had to remove `should_` because of the new pre-commit hook. That's created a bit of noise in this PR, so apologies to the reviewers!

## Guidance to review

Easier to review commit by commit

## Link to Trello card

[Trello card](https://trello.com/c/VMAmtMQe)

### CODE REVIEW

**Comment by elceebee** [comment]
[Apply: Implement small content improvements](https://trello.com/c/VMAmtMQe/728-apply-implement-small-content-improvements)

**Review by tomas-stefano — APPROVED** [review]
🚀

File: app/views/candidate_mailer/nudge_unsubmitted_with_incomplete_courses.text.erb
Comment (RebeccaCryan) [inline app/views/candidate_mailer/nudge_unsubmitted_with_incomplete_courses.text.erb:1]: and same here with `you'd` changing to `you would`

File: spec/components/candidate_interface/application_dashboard_course_choices_component_spec.rb
Comment (RebeccaCryan) [inline spec/components/candidate_interface/application_dashboard_course_choices_component_spec.rb:1]: and here for `it's` and `it is`

File: app/views/candidate_interface/guidance/index.html.erb
Comment (RebeccaCryan) [inline app/views/candidate_interface/guidance/index.html.erb:1]: Can we change `We'll` to `We will`?

File: app/views/provider_mailer/courses_open_on_apply.text.erb
Comment (RebeccaCryan) [inline app/views/provider_mailer/courses_open_on_apply.text.erb:1]: you will also get a notification email when you receive any applications.
The `y` should be a capital `Y`

File: spec/components/provider_interface/diversity_information_component_spec.rb
Comment (RebeccaCryan) [inline spec/components/provider_interface/diversity_information_component_spec.rb:1]: you will be able to view this if the candidate accepts an offer for this application
The `y` needs to be upper case `Y`

File: spec/components/candidate_interface/gcse_equivalency_required_component_spec.rb
Comment (RebeccaCryan) [inline spec/components/candidate_interface/gcse_equivalency_required_component_spec.rb:1]: Should GCSEs be capitalised as an acronym?

Comment (elceebee) [reply spec/components/candidate_interface/gcse_equivalency_required_component_spec.rb:1]: Fortunately, that's not text that is seen by anyone but developers.

**Review by tomas-stefano — APPROVED** [review]

**Pull request merged**